src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -330,8 +332,7 @@ private void EventLoop()
                             _handleToContextMap.TryGetValue(handle, out SocketAsyncContext context);
                             if (context != null)
                             {
-                                context.HandleEvents(_buffer[i].Events);
-                                context = null;
+                                _ioQueue.Schedule(context, _buffer[i].Events);
```

Comment:
`context` is set to `null` here because the scope of `context` is larger than you'd expect, and causes the last used `context` to remain rooted. There's a test that needs to be set to `null`.